plugins/parquet/parquetembed.cpp

```diff
@@ -1895,12 +1904,64 @@ void ParquetRecordBinder::processReal(double value, const RtlFieldInfo *field)
         failx("Incorrect type for real field %s: %s", field->name, fieldBuilder->type()->ToString().c_str());
 }
 
+/**
+ * @brief Adds DECIMAL and UDECIMAL fields to the builder based on the size of the decimal
+ *
+ * @param decText Data to be written to the Parquet file.
+ * @param bytes Number of bytes holding the digits.
+ * @param digits Number of digits in the decimal.
+ * @param precision Number of digits after the decimal point.
+ * @param field RtlFieldInfo holds metadata about the field.
+ */
+void ParquetRecordBinder::addDecimalFieldToBuilder(rtlDataAttr *decText, size32_t bytes, int32_t digits, int32_t precision, const RtlFieldInfo *field)
+{
+    arrow::ArrayBuilder *fieldBuilder = parquetWriter->getFieldBuilder(field);
+    if (fieldBuilder->type()->id() == arrow::Type::DECIMAL128)
+    {
+        arrow::Decimal128Builder *decimal128Builder = static_cast<arrow::Decimal128Builder *>(fieldBuilder);
+        arrow::Decimal128 decimal128;
+        reportIfFailure(arrow::Decimal128::FromString(std::string_view(decText->getstr(), bytes), &decimal128, &digits, &precision));
+        reportIfFailure(decimal128Builder->Append(decimal128));
+    }
+    else if (fieldBuilder->type()->id() == arrow::Type::DECIMAL256)
+    {
+        arrow::Decimal256Builder *decimal256Builder = static_cast<arrow::Decimal256Builder *>(fieldBuilder);
+        arrow::Decimal256 decimal256;
+        reportIfFailure(arrow::Decimal256::FromString(std::string_view(decText->getstr(), bytes), &decimal256, &digits, &precision));
+        reportIfFailure(decimal256Builder->Append(decimal256));
+    }
+    else
+        failx("Incorrect type for Decimal field %s: %s", field->name, fieldBuilder->type()->ToString().c_str());
+}
+
+/**
+ * @brief Calls processDecimal since unsigned decimals still have a max of 64 digits with 32 digits of precision.
+ *
+ * @param value Data to be written to the Parquet file.
+ * @param digits Number of bytes holding the digits.
```

Comment:
Minor: An argument named 'digits' makes me think "number of base-10 digits in the value." Perhaps I am the only one that thinks like that but if not: Renaming that argument for clarity may be in order.

Reply:
Previous comment replicated to processDecimal() as well.

Reply:
I agree with your comment as I expected the same thing. The digits name is defined in the IFieldProcessor Interface. Would this require renaming all instances of digits in the code where IFieldProcessor is implemented?

Reply:
Ah, never mind then. If this was purely local to your code then the rename may be in order. If it is part of the de facto standard in the rest of the code then it's better that it stays this way. Thanks!